OrbitCore/Params.cpp

```diff
@@ -62,19 +62,28 @@ ORBIT_SERIALIZE(Params, 19) {
 void Params::Save() {
   GCoreApp->SendToUiNow("UpdateProcessParams");
   std::string fileName = Path::GetParamsFileName();
-  SCOPE_TIMER_LOG(absl::StrFormat("Saving params in %s", fileName.c_str()));
+  SCOPE_TIMER_LOG(absl::StrFormat("Saving Params in \"%s\"", fileName.c_str()));
   std::ofstream file(fileName);
   cereal::XMLOutputArchive archive(file);
   archive(cereal::make_nvp("Params", *this));
 }
 
 //-----------------------------------------------------------------------------
 void Params::Load() {
-  std::ifstream file(Path::GetParamsFileName());
-  if (!file.fail()) {
+  std::string filename = Path::GetParamsFileName();
+  std::ifstream file(filename);
+  if (file.fail()) {
+    // Create the file with default values if it doesn't exist.
+    Save();
```

Comment:
Can Save() not throw any exceptions? (It also uses cereal.)

Reply:
It's a library we don't control, it could always throw exceptions. Does that mean we should proactively guard all calls to external libraries?

Reply:
not every library throws exceptions. but for those which do - we should probably guard all calls. maybe create a proxy for that.